### ISSUE TITLE
Provide nvda with the possibility to navigate tables in google docs with the use of nvda shortcut 

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#7942]
### Steps to reproduce:

Open a new document in google docs.
Create a new table with 2 rows and 2 columns.
Now fill in all the cells of the table.
Now being on the first column and row of the table press nvda + space to go to browse mode .
Now press ctrl + alt + arrows
In my tests, nvda may sometimes be able to navigate through some rows or columns, but will never be able to navigate the entire table, at some point even though it is on the table it will inform you that it is outside the table.
Braille support is enabled for testing.
With braille support disabled when I put it in browse mode , I could not find the table.

### Expected behavior:
When placing nvda in browse mode , you would expect nvda to navigate the entire table using the nvda navigation keys with alt + ctrl + arrows.
### Actual behavior:
The nvda using the table navigation keys for sometimes walks between some cells, but never can explore the entire table without even being on a report cell that is not in a table.
### System configuration:
NVDA version:
> Example: next-14806
NVDA Installed or portable:
instaled
Other information:
If there is any way for nvda to support navigating tables using ctrl + alt + arrows without having to pause browsing mode would be particularly facilitating in google docs.

Windows version:
> Example: Windows 10 Version 1709 16299.192 64bits

Name and version of other software in use when reproducing the issue:
chrome 64 64bits
### Other questions:

Does the issue still occur after restarting your PC?
yes
Have you tried any other versions of NVDA?
> Please list them and the result
no

**Comment by PratikP1** [comment]
The ideal behavior to navigate tables should not require the user to go to browse mode in Docs. It is an app and should act as such. Table navigation should occur as if the user were in a Word document.

**Comment by leonardder** [comment]
@fernando-jose-silva: could you please update your description using the right terminology, browse mode and focus mode? It is not clear what you mean with exploration mode and navigation mode. I assume you mean browse mode with exploration mode and focus mode with navigation mode, but it would be nice if we'd be sure.

**Comment by fernando-jose-silva** [comment]
Corrected the terminologies.
I did not pay attention on how the google translator had translated into English.
My original request would be to be in browse mode that navigation through the table would be performed without errors.
As @ PratikP1 commented I would like to stress that if possible to use the navigation commands in tables like ctrl + alt + arrows while also in focus mode would be very interesting.

**Comment by leonardder** [comment]
Could you reproduce this with the most recent next snapshot, which should support table navigation in focus mode for Google Docs?

**Comment by michaelDCurran** [comment]
Note it seems you'll have to use Firefox 57 or higher to test this, as 
Firefox 52 has a broken table implementation in Google Docs for some reason.

**Comment by fernando-jose-silva** [comment]
Thank you very much for the implemented functionality, with me it worked.
I would like to make some behavior notes that I found that if possible could be improved, however these problems are minor and do not dissapoint neither one nor the new functionality.
Open a new document in google docs and create a table with two rows and two columns.
Now position in the first row and first column.
Enter a content in the first row and column in my tests typed test1.
Now press tabe to position in row 1 column 2.
Real:
the nvda is mute.
expected:
Nvda should announce that you have been positioned in row 1 column 2 and that the cell is empty.
Now finish filling out the worksheet, for the tests I filled in like this:
line 1 column 1 test1
line 1 column 2 test2
line 2 column 1 test3
line 2 column 2 test4
Now position yourself in column 1 row 1.
Press ctrl + alt + right arrow.
You will be in column 2 row 1 as expected.
Now press ctrl + alt + down arrow you will be in row 2 column 2 as expected.
Press ctrl + alt + up arrow you will be in row 1 column 2 as expected.
Press ctrl + alt + left arrow and you will be in row 1 column 1 as expected.
And finally, press ctrl + alt + arrow below:
real:
you will be in row 2 column 2.
Expected:
you should be in column 1 row 2.
Sorry for the long way, but it was the only way to reliably reproduce the error on my machine.
Stop that after moving several times in the worksheet by the cells using the shortcut keys someone failed or nvda or google docs and the focus does not stay where you expected.
firefox 58.0.1 64bit
windows 10 16.299.192.

**Comment by Adriani90** [comment]
@fernando-jose-silva thanks for reporting. I will close this issue as works for me because it is solved. For your other issues regarding tables, please reopen new issues. Thanks!

**Comment by Adriani90** [comment]
After testing further in Chrome 74, when switching to focus mode in google docs and navigating through a table, I get following error and NVDA does not speak at all:

```
IO - inputCore.InputManager.executeGesture (13:54:21.397):
Input: kb(laptop):alt+control+downArrow
ERROR - scriptHandler.executeScript (13:54:21.477):
error executing script: <bound method Dynamic_EditableTextWithAutoSelectDetectionEditorIa2WebIAccessible.script_nextRow of <NVDAObjects.Dynamic_EditableTextWithAutoSelectDetectionEditorIa2WebIAccessible object at 0x058862B0>> with gesture u'Alt+Strg+Pfeiltaste nach unten'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 187, in executeScript
  File "documentBase.pyc", line 174, in script_nextRow
  File "documentBase.pyc", line 161, in _tableMovementScriptHelper
  File "documentBase.pyc", line 126, in _getNearestTableCell
TypeError: coercing to Unicode: need string or buffer, int found
```

**Comment by Adriani90** [comment]
tested with NVDA alpha-17099,0d2a1525 and Chrome 74.

**Comment by Adriani90** [comment]
this is solved now in NVDA alpha-17240,c20a5032 by PR #9562. Closing as works for me. @fernando-jose-silva if you are still having this issue in last NVDA alpha, please comment on it and we can reopen the discussion. thanks.
